docs/bucket-notifications/README.md

```diff
@@ -0,0 +1,435 @@
+import pika
+
+connection = pika.BlockingConnection(pika.ConnectionParameters(
+        host='localhost'))
+channel = connection.channel()
+
+channel.exchange_declare(exchange='bucketevents',
+                         type='fanout')
+
+result = channel.queue_declare(exclusive=False)
+queue_name = result.method.queue
+
+channel.queue_bind(exchange='bucketevents',
+                   queue=queue_name)
+
+print(' [*] Waiting for logs. To exit press CTRL+C')
+
+def callback(ch, method, properties, body):
+    print(" [x] %r" % body)
+
+channel.basic_consume(callback,
+                      queue=queue_name,
+                      no_ack=False)
+
+channel.start_consuming()
+```
+
+
+Execute this example python script to watch for RabbitMQ events on the console.
+
+```py
+python rabbit.py
+```
+
+Open another terminal and upload a JPEG image into “images” bucket.
+
+```
+mc cp myphoto.jpg myminio/images
+```
+
+You should receive the following event notification via RabbitMQ once the upload completes.
+
+```py
+python rabbit.py
+‘{“Records”:[{“eventVersion”:”2.0",”eventSource”:”aws:s3",”awsRegion”:”us-east-1",”eventTime”:”2016–09–08T22:34:38.226Z”,”eventName”:”s3:ObjectCreated:Put”,”userIdentity”:{“principalId”:”minio”},”requestParameters”:{“sourceIPAddress”:”10.1.10.150:44576"},”responseElements”:{},”s3":{“s3SchemaVersion”:”1.0",”configurationId”:”Config”,”bucket”:{“name”:”images”,”ownerIdentity”:{“principalId”:”minio”},”arn”:”arn:aws:s3:::images”},”object”:{“key”:”myphoto.jpg”,”size”:200436,”sequencer”:”147279EAF9F40933"}}}],”level”:”info”,”msg”:””,”time”:”2016–09–08T15:34:38–07:00"}\n
+```
+<a name="Elasticsearch"></a>
+## Publish Minio events via Elasticsearch 
+
+Install Elasticsearch 2.4 from [here](https://www.elastic.co/downloads/past-releases/elasticsearch-2-4-0).
+
+## Recipe steps
+
+### Step 1: Add Elasticsearch endpoint to Minio
+
+The default location of Minio server configuration file is ``~/.minio/config.json``. Update the Elasticsearch configuration block in ``config.json`` as follows:
+
+```
+"elasticsearch": {
+                        "1": {
+                                "enable": true,
+                                "url": "http://127.0.0.1:9200",
+                                "index": "bucketevents"
+                        }
+                },
+```
+Restart Minio server to reflect config changes. ``bucketevents`` is the index used by Elasticsearch.
+
+If you are running [distributed Minio](https://docs.minio.io/docs/distributed-minio-quickstart-guide), modify ``~/.minio/config.json`` with these local changes on all the nodes.
+
+### Step 2: Enable bucket notification using Minio client
+
+We will enable bucket events only when JPEG images are uploaded or deleted from ``images`` bucket on ``myminio`` server. Here ARN value is ``arn:minio:sqs:us-east-1:1:elasticsearch``. To understand more about ARN please follow [AWS ARN](http://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) documentation.
```

Comment:
Same changes in this line as suggested for RabbitMQ.
 - We will enable bucket event notification to trigger whenever a JPEG image is uploaded or deleted ...